### DISCUSSION TITLE
Show custom POI set (speed cameras)

### DISCUSSION THREAD

**Discussion by JimmyS83** [devemux86/cruiser#222] [Q&A] [answered]
Hi,

another thing comes to my mind. Would be nice to allow import POI in some common standard and have option to show them or not.. That would be especially usefull for speed camera DB's, which are publicly available. I understand that POI system isnt easy and its complex, but take this just as kickoff idea. On motorcycle, two most important POIs which I am using for display/search are speed cameras and petrol stations. I think there would be even tools for creating/converting POI files..

I understand I can download big package of POIs from mapforge, but I didnt find other way to utilize them except search - I mean option to show some of them them in Map or something like that. (and I assume speed cameras are not in the bundle?)

On the same note, is it me, or Max speed icon isnt displayed on the top of application in follow mode? Or I must break speed limit to show it?  (I am used that speed limit is shown all the time from other app, so I was a bit curious, how to get the speed limit signs as you have in pictures :)

**Comment by devemux86** [comment, marked as answer]
@JimmyS83 thanks for the suggestion.

> Would be nice to allow import POI in some common standard.

You can import external POI via gpx in Cruiser as favorites into separate groups with custom icons / colors and visibility:
- https://github.com/devemux86/cruiser/discussions/45#discussioncomment-5791140

Nearby favorties in navigation will also be available really soon:
- https://github.com/devemux86/cruiser/discussions/209

> Max speed icon isn't displayed on the top of application in follow mode?

Speed limits are provided by the calculated / imported routes.
Only a few services (like GraphHopper / Kurviger) have speed limits in their routes.

**Comment by devemux86** [comment]
> I think there would be even tools for creating/converting POI files..

That's why [I implemented Mapsforge offline POI](https://github.com/mapsforge/mapsforge/blob/master/docs/POI.md) several years ago! :slightly_smiling_face: 

> I understand I can download big package of POIs from mapforge, but I didnt find other way to utilize them except search - I mean option to show some of them them in Map or something like that.

Offline POI will soon have new features, like display offline POi categories on the map.

> I assume speed cameras are not in the bundle?

Speed cameras are already included in offline POI, it will be available in the nearby POI in navigation:
- https://github.com/devemux86/cruiser/discussions/9#discussioncomment-6249887

Currently included offline POI can be seen here:
https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml

If you need more POI please let me know and I can enable them.

**Comment by JimmyS83** [comment]
Thats all great news, thanks!  I will try to learn some favourites, and definitivelly look up on future updates! then :)

**Reply by devemux86** [reply]
@JimmyS83 

Please explore the new favorites, one of Cruiser's latest powerful features with many customization options.

If you have any questions or ideas for improvements, please let me know! :slightly_smiling_face:

**Comment by JimmyS83** [comment]
I tried favourites, and yes, it works :) The proximity popup in future version will be great, also native use of .poi file itself, and maybe use of direction note, but meanwhile, if anyone found this useful, I just:

Filter desired POIs (143 as example for speed camera)

```
SELECT i.lat, i.lon
FROM poi_category_map c
INNER JOIN poi_index i
ON c.id = i.id
INNER JOIN poi_data d
ON c.id = d.id
WHERE c.category = 143
```

export it to CSV, conver CSV to GPX (https://mygeodata.cloud/converter/csv-to-gpx), and then just import one file into Cruiser :+1: :
![photo_2023-06-23_09-32-41](https://github.com/devemux86/cruiser/assets/7437057/102ec36a-579c-4bdb-ba64-60ddc8a0409d)

As was already mentioned in Discussion, I agree that camera icon is more for Tourist place :), but there is no "usual" speed camera icon yet, not a big deal.

**Reply by devemux86** [reply]
@JimmyS83 

You could also use the [overpass-turbo](https://overpass-turbo.eu/) to query the elements in a region and export them.
OpenStreetMap Wiki provides overpass-turbo queries in all tag pages.

For example about speed cameras:
- [OpenStreetMap Wiki](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dspeed_camera)
- [overpass-turbo](https://overpass-turbo.eu/?template=key-value&key=highway&value=speed_camera)

**Comment by Ciutnik** [comment]
@devemux86 
> If you need more POI please let me know and I can enable them.

Is there a chance that some polygons will be visible as a POI? A long time ago, on the previous forum, we talked about such cases. E.g. `building=ruins` or `historic=castle`.

**Reply by devemux86** [reply]
Do you mean on the map as symbols or in the poi database to search?

historic=castle and historic=ruins exist in both maps + poi.
See the [tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml) for maps and the [poi-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml) for poi for all the currently included tags.

Also checking the [Biker theme](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/biker.xml), only the castles are rendered with the "Tourist attractions" layer.

**Comment by Ciutnik** [comment]
Ah indeed, sorry, the polygons of the castles are visible as POIs, but the ruins (polygons of the buildings) are not. So are museums or cathedrals. I once modified Kurviger's default map theme so that I could see these objects on the map. But let me wait with requests until advanced POI support in Cruiser 🙂

**Reply by devemux86** [reply]
So you mean the poi files for offline search?

Currently the [poi-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml) does not contain the "building=" as key.

**Comment by Ciutnik** [comment]
> So you mean the poi files for offline search?

Rather, I am more interested in the **visibility** of points on the map. When planning a trip, I like to browse the map in search of interesting points. In Osmand, I can define the view of the above-mentioned objects on the map. It's cleverly organized.

> Currently the poi-mapping does not contain the "building=" as key.

I don't remember how I managed to achieve this (as soon as themes appeared in K2), but I forced this behavior by modifying the default one.

**Reply by devemux86** [reply]
@Ciutnik so you mean the map, we need to clarify this. :slightly_smiling_face: 

Map [tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml) seems to contain both tags.
[Biker](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/biker.xml) / [MotoRider](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/motorider.xml) themes only have the castles in "Tourist attractions".

Can you make the needed changes in the theme for the ruins too and see that it works?
Then I can update the themes.

**Reply by Ciutnik** [reply]
> so you mean the map, we need to clarify this.

Of course, when I say "map" in the planning mode, I mean the road network and clearly visible POIs. Building polygons are possibly cool when zoomed in, so they're of little use for planning. Therefore, I would like attractive objects that are mapped as polygons to be marked as POIs. As I mentioned, OsmAnd does this very well.

**Reply by devemux86** [reply]
OsmAnd works very well, it can search and show POI on the map as overlays which can have touch events.

This will come in Cruiser with the new categorized search. It is the final piece to complete the offline search:
- https://github.com/devemux86/cruiser/discussions/74
